src/main/java/org/kiwiproject/test/mongo/MongoTestProperties.java

```diff
@@ -92,30 +96,34 @@
      * <p>
      * Use the fluent builder as an alternative to this constructor.
      *
-     * @param hostName           the host where MongoDB is located
-     * @param hostDomainBehavior how to handle domains in the given {@code hostName}
-     *                           (defaults to {@link HostDomainBehavior#STRIP STRIP} if this argument is {@code null})
-     * @param port               the port that MongoDB is listening on
-     * @param serviceName        the name of the service/application being tested
-     * @param serviceHost        the host of the service/application being tested
+     * @param hostName          the host where MongoDB is located
+     * @param port              the port that MongoDB is listening on
+     * @param serviceName       the name of the service/application being tested
+     * @param serviceHost       the host of the service/application being tested
+     * @param serviceHostDomain how to handle domains in the given {@code serviceHost}
+     *                          (defaults to {@link ServiceHostDomain#STRIP STRIP} if this argument is {@code null})
      */
     @Builder
     public MongoTestProperties(String hostName,
-                               @Nullable HostDomainBehavior hostDomainBehavior,
                                int port,
                                String serviceName,
-                               String serviceHost) {
+                               String serviceHost,
+                               @Nullable ServiceHostDomain serviceHostDomain) {
         this.hostName = hostName;
```

Comment:
Consider adding `KiwiPreconditions.requireNotBlank` on `hostName`, `serviceName`, and `serviceHost`